src/main/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseController.java

```diff
@@ -359,6 +323,55 @@ public ResponseEntity<CaseListResponse> getCaseList(
                 .body(CaseListResponse.builder().cases(courtCaseResponses).build());
     }
 
+    @Operation(summary = "Gets case data for a court on a date and parameters",
+            description = "Response is sorted by court room, session start time and by defendant surname. The createdAfter and " +
+                    "createdBefore filters will not filter out updates originating from prepare-a-case, these manual updates" +
+                    " are always assumed to be correct as they have been deliberately made by authorised users rather than " +
+                    "automated systems.")
+    @GetMapping(value = "/court/{courtCode}/matcher-cases", produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<CaseListResponse> getCaseListForMatcher(
+            @PathVariable String courtCode,
+            @RequestParam(value = "date")
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date,
+            @RequestParam(value = "numberOfPossibleMatches", required = false) Long numberOfPossibleMatches,
+            @RequestParam(value = "forename", required = false) String forename,
+            @RequestParam(value = "surname", required = false) String surname,
+            @RequestParam(value = "defendantName", required = false) String defendantName,
+            @RequestParam(value = "caseId", required = false) String caseId,
+            @RequestParam(value = "hearingId", required = false) String hearingId,
+            @RequestParam(value = "defendantId", required = false) String defendantId,
+            WebRequest webRequest
+    ) {
+        var partialResponse = ResponseEntity.ok();
+        if (enableCacheableCaseList) {
+            var lastModified = courtCaseService.filterHearingsLastModified(courtCode, date)
+                    .orElse(NEVER_MODIFIED_DATE)
+                    .toInstant(ZoneOffset.UTC);
+            if (webRequest.checkNotModified(lastModified.toEpochMilli())) {
+                return ResponseEntity.status(HttpStatus.NOT_MODIFIED)
+                        .cacheControl(CacheControl.maxAge(MAX_AGE, TimeUnit.SECONDS))
+                        .build();
+            }
+
+            partialResponse = partialResponse
+                    .lastModified(lastModified)
+                    .cacheControl(CacheControl.maxAge(MAX_AGE, TimeUnit.SECONDS));
+        }
+
+        final var hearingSearchFilter = HearingSearchFilter.builder()
+                .courtCode(courtCode)
+                .hearingDay(date)
+                .source(null)
+                .breach(false)
+                .build();
+
+        var courtCases = courtCaseService.filterHearings(hearingSearchFilter);
+
+        var courtCaseResponses = filterCourtCaseResponses(date, numberOfPossibleMatches, forename, surname, defendantName, caseId, hearingId, defendantId, courtCases);
```

Comment:
The previous block of code builds up a filter object to pass into the filter method, rather than passing in a lot of arguments - might be more readable to do that here too?

It also delegates the filtering to another class rather than doing it in the controller

Reply:
Agreed, moved the logic down to the service and used the filter wrapper